### PR TITLE
update to the latest package:dart_flutter_team_lints

### PR DESCRIPTION
**Pull request by devoncarew**
- update to the latest `package:dart_flutter_team_lints`

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/example/sample_rate.dart [hunk 1]
```diff
@@ -7,15 +7,15 @@ import 'package:unified_analytics/src/utils.dart';
 /// The purpose of this example file is to demonstrate the sampling
 /// rate functionality from the survey handler.
 ///
-/// It defines a [remoteUniqueId] that subs in for a real survey's unique
```
Comment (eliasyishak) [inline pkgs/unified_analytics/example/sample_rate.dart:10]: Seems like the lint removes some of the brackets from the dartdocs while keeping them in other places, was that intended? @devoncarew

File: pkgs/unified_analytics/lib/src/ga_client.dart [hunk 2]
```diff
@@ -50,11 +50,11 @@ class GAClient {
   /// Receive the payload in Map form and parse
   /// into JSON to send to GA.
   ///
-  /// The [Response] returned from this method can be
+  /// The [http.Response] returned from this method can be
```
Comment (eliasyishak) [inline pkgs/unified_analytics/lib/src/ga_client.dart:53]: Example of where it was kept

File: pkgs/unified_analytics/example/sample_rate.dart (hunk 1 above)
Comment (devoncarew) [reply pkgs/unified_analytics/example/sample_rate.dart:10]: Yes - this is due to adding the `comment_references` lint to the latest version of `package:dart_flutter_team_lints`. We're now validating that dartdoc references can be resolved.

Where these resolved - or I could update the reference to point to a valid symbol - I kept the square brackets. Other places I removed the brackets or changed them to just use regular code formatting (back-ticks).

Comment (devoncarew) [reply pkgs/unified_analytics/example/sample_rate.dart:10]: Here, we're not able to refer to local variable references within the method.

Comment (eliasyishak) [reply pkgs/unified_analytics/example/sample_rate.dart:10]: Ah got it, that's a very helpful lint! LGTM

**Pull request merged**